### PR TITLE
First iteration of pie charts in d3.

### PR DESCRIPTION
**Pull request by lindsaycarr**
Working on #71. Not sure if this counts for "basic transitions" or not. Someone else can take over from here for next week.

The data reformatting should probably happen elsewhere, but I'm not exactly sure how to make that happen in R. We need an array where you can point to one "column" where all of the values for determining pie slice size are.

E.g. right now you get the size of the circle by saying `d.thermoelectric`, but we need to be able to say `d.value` and have that apply to all water use types.

If we want windshield wiper transitions and follow my idea in #72 where we just make all of them pie charts, then I think we can just format the data this way in R and not worry about it. I guess we could also format it this way and change the way we are calling it for individual categories, too.

Transitions between pie charts and other categories need work. Tried to make them smooth, but that didn't seem to do anything. I also tried to distinguish transitions between regular categories vs a single category to pie or pie to a single category. You'll see that effort in the function `updateCategory`.

Performance issue: as of now, it appears that every time you click or hover on category, completely new pie groups are being added rather than updated. The goal would to only have a total of 2x number of counties in the group with id `map` (one set of circles, one set of pie groups). But it just keeps growing if you keep changing categories. I've made note of this in #82.

piecharts

### CODE REVIEW

**Comment by lindsaycarr** [comment]
Changed "other" category to purple for now so people could actually see that category.

image

File: js/pie_utils.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    .value(function(d) { return d.value; });
+
+var arcpath = d3.arc()
+    .innerRadius(0);
+
+function addPieCharts() {
+  
+  // relies on map and countyCentroids as global vars
+   
+  var pieformdata = pieData(countyCentroids);
```
[the remark this answers is not in the document]
Comment (lindsaycarr) [reply js/pie_utils.js]: Yes, that would be ideal!

[the remark this answers is not in the document]
Comment (aappling-usgs) [reply js/pie_utils.js]: could you frame this as a new issue with an example of the data you need, in such a way that a non-D3 person could take on the task?

[the remark this answers is not in the document]
Comment (aappling-usgs) [reply js/pie_utils.js]: I'm planning for us to drop the pie charts from the interactive vizzy, so let's hold off until we know the data format we'll need.

**Pull request merged**